### PR TITLE
Fix directory traversal bug (replacement of #574)

### PR DESCRIPTION
**Pull request by nmantani**
Hi,

I improved handling of Windows UNC paths of normalize() in qiling/os/utils.py
and added tests/test_pathutils.py for test of PathUtils functions.
I also updated Changelog.

This pull request fixes the directory traversal bug reported in #572 and replaces the pull request #574 .

### CODE REVIEW

**Comment by wtdcode** [comment]
Hi!

Glad to see your PR. I will review it tomorrow.

File: qiling/os/utils.py [hunk 1]
```diff
@@ -78,19 +78,53 @@ def convert_posix_to_win32(rootfs, cwd, path):
...
         if path.is_absolute():
-            return rootfs / path.relative_to(path.anchor)
+            return rootfs / PathUtils.normalize(path)
         else:
-            return rootfs / cwd / path
+            return rootfs / PathUtils.normalize(cwd / path.as_posix())
+
+    @staticmethod
+    def normalize(path):
+        sep = path._flavour.sep
```
Comment (wtdcode) [inline qiling/os/utils.py:97]: Quite tricky here. But it seems that we can't find a better approach. I suggest adding a dict like:

```python
seps = {
PureWindowsPath : "\\"
...
}
```

File: qiling/os/utils.py [hunk 2]
```diff
@@ -78,19 +78,53 @@ def convert_posix_to_win32(rootfs, cwd, path):
...
+
+        for name in str(path).split(sep):
+            if not name or name == '.': # current directory
+                continue
+
+            if name == '..': # parent directory
+                (normalized_path, _, _) = normalized_path.rpartition(sep)
+                continue
+
+            normalized_path += sep + name
```
Comment (wtdcode) [inline qiling/os/utils.py:112]: Why not pathlib here? Manual path concatenation often leads to critical bugs.

**Review by wtdcode — CHANGES_REQUESTED** [review]

**Comment by wtdcode** [comment]
Note that please also resolve the conflicts.

**Comment by nmantani** [comment]
Thank you for your review! I resolved conflict of Changelog.
I will rewrite normalize() function based on your advice.

**Comment by nmantani** [comment]
I have rewritten normalize() function and manual path concatenation is eliminated.
Please check it.

**Comment by wtdcode** [comment]
Looks good now. Thanks for your fix!

**Pull request merged**